sensing/radar_threshold_filter/src/radar_threshold_filter_node/radar_threshold_filter_node.cpp

```diff
@@ -0,0 +1,186 @@
+  node_param_.is_range_filter = declare_parameter<bool>("node_params.is_range_filter", false);
+  node_param_.range_min = declare_parameter<double>("node_params.range_min", 0.0);
+  node_param_.range_max = declare_parameter<double>("node_params.range_max", 0.0);
+  node_param_.is_azimuth_filter = declare_parameter<bool>("node_params.is_azimuth_filter", false);
+  node_param_.azimuth_min = declare_parameter<double>("node_params.azimuth_min", 0.0);
+  node_param_.azimuth_max = declare_parameter<double>("node_params.azimuth_max", 0.0);
+
+  // Subscriber
+  sub_radar_ = create_subscription<RadarScan>(
+    "~/input/radar", rclcpp::QoS{1}, std::bind(&RadarThresholdFilterNode::onData, this, _1));
+
+  // Publisher
+  pub_radar_ = create_publisher<RadarScan>("~/output/radar", 1);
+
+  // Timer
+  const auto update_period_ns = rclcpp::Rate(node_param_.update_rate_hz).period();
+  timer_ = rclcpp::create_timer(
+    this, get_clock(), update_period_ns, std::bind(&RadarThresholdFilterNode::onTimer, this));
+}
+
+void RadarThresholdFilterNode::onData(const RadarScan::ConstSharedPtr msg) { radar_data_ = msg; }
+
+rcl_interfaces::msg::SetParametersResult RadarThresholdFilterNode::onSetParam(
+  const std::vector<rclcpp::Parameter> & params)
+{
+  rcl_interfaces::msg::SetParametersResult result;
+  try {
+    {
+      auto & p = node_param_;
+      update_param(params, "node_params.update_rate_hz", p.update_rate_hz);
+      update_param(params, "node_params.is_amplitude_filter", p.is_amplitude_filter);
+      update_param(params, "node_params.amplitude_min", p.amplitude_min);
+      update_param(params, "node_params.amplitude_max", p.amplitude_max);
+      update_param(params, "node_params.is_range_filter", p.is_range_filter);
+      update_param(params, "node_params.range_min", p.range_min);
+      update_param(params, "node_params.range_max", p.range_max);
+      update_param(params, "node_params.is_azimuth_filter", p.is_azimuth_filter);
+      update_param(params, "node_params.azimuth_min", p.azimuth_min);
+      update_param(params, "node_params.azimuth_max", p.azimuth_max);
+    }
+  } catch (const rclcpp::exceptions::InvalidParameterTypeException & e) {
+    result.successful = false;
+    result.reason = e.what();
+    return result;
+  }
+  result.successful = true;
+  result.reason = "success";
+  return result;
+}
+
+bool RadarThresholdFilterNode::isDataReady()
+{
+  if (!radar_data_) {
+    RCLCPP_INFO_THROTTLE(get_logger(), *get_clock(), 1000, "waiting for radar msg...");
+    return false;
+  }
+
+  return true;
+}
+
+void RadarThresholdFilterNode::onTimer()
+{
+  if (!isDataReady()) {
+    return;
+  }
+
+  RadarScan output;
+  output.header = radar_data_->header;
+  for (const auto & radar_return : radar_data_->returns) {
+    if (isWithinThreshold(radar_return)) {
+      output.returns.push_back(radar_return);
+    }
+  }
+
+  pub_radar_->publish(output);
+}
```

Comment:
Please let me know if there is a reason why you are publishing onTimer.
Basically, I think it is better to publish with onData.

Reply:
Fixed at [2824cb0](https://github.com/autowarefoundation/autoware.universe/pull/2300/commits/2824cb07ab6915b1d2f311b017f5cf5dbc83dd64)